src/main/c/Objects/pyjclass.c

```diff
@@ -134,17 +136,63 @@ static PyObject* pyjclass_add_inner_classes(JNIEnv *env,
     return (PyObject*) topClz;
 }
 
+/*
+ * Populate an attrs dict with all the Java methods and fields for the given
+ * class. This is currently the only way static methods and fields are made
+ * available. There is no known use for non-static fields but for now they are
+ * left in for backwards compatiblity, consider removing them in the future.
+ */
+static PyObject* pyjclass_init_attr(JNIEnv *env, jclass clazz)
+{
+    PyObject *attr = PyDict_New();
+    if (!attr) {
+        return attr;
+    }
+
+    if (!(*env)->IsAssignableFrom(env, clazz, JOBJECT_TYPE)) {
+        /*
+         * Don't bother with primitive types, they don't work with PyJType and they
+         * don't have any fields or methods.
+         */
+        return attr;
+    }
+    PyTypeObject* type = PyJType_Get(env, clazz);
+    if (!type) {
+        Py_DecRef(attr);
+        return NULL;
+    }
+    PyObject *key, *value;
+    Py_ssize_t pos = 0;
+    while (PyDict_Next(type->tp_dict, &pos, &key, &value)) {
```

Comment:
I don't understand how this while loop works.  Are all the fields and methods in tp_dict by this point?

Reply:
Yes, all the reflection happens before the type is created so we can just copy from the type. I added a comment to the function to clarify what it does.